mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -222,134 +280,423 @@ public void testSimpleAttachment3() throws MessagingException, IOException {
+        //Then
+        assertThat(actual).isFalse();
+    }
 
-        Assert.assertTrue(mail.getMessage().getContent() instanceof MimeMultipart);
-        Assert.assertEquals(2, ((MimeMultipart) mail.getMessage().getContent())
-                .getCount());
+    @Test
+    public void analyseMultipartPartMessageShouldRemovePartWhenOnePartShouldHaveBeenRemoved() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
 
-        Object actual = ((MimeMultipart) mail.getMessage().getContent())
-                .getBodyPart(1).getContent();
-        if (actual instanceof ByteArrayInputStream) {
-            Assert.assertEquals(body2, toString((ByteArrayInputStream) actual));
-        } else {
-            Assert.assertEquals(body2, actual);
-        }
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("remove", "all")
+                .setProperty("pattern", ".*")
+                .build();
+        mailet.init(mci);
+        MimeMultipart multiPart = new MimeMultipart();
+        
+        MimeBodyPart part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part.setFileName("removeMe.tmp");
+        multiPart.addBodyPart(part);
+        MimeMessage message = new MimeMessage(Session
+                .getDefaultInstance(new Properties()));
+        message.setContent(multiPart);
+        message.saveChanges();
+        Mail mail = mock(Mail.class);
+        //When
+        mailet.analyseMultipartPartMessage(message, mail);
+        //Then
+        assertThat(multiPart.getCount()).isZero();
+    }
 
+    @Test
+    public void saveAttachmentShouldUsePartNameIfNoFilename() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        Part part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part.setFileName("example.tmp");
+        String fileName = null;
+        //When
+        String actual = mailet.saveAttachmentToFile(part, fileName);
+        //Then
+        assertThat(actual).startsWith("example");
+        assertThat(actual).endsWith(".tmp");
     }
+    
+    @Test
+    public void saveAttachmentShouldThrowWhenNoFilenameAtAll() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        Part part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        String fileName = null;
 
+        expectedException.expect(NullPointerException.class);
```

Comment:
Nope, you don't understand me.

As an admin I don't care too much about the ICAL being sent to OpenPaas, but I care about users receiving mails. Your mailet SHOULD NOT fail on single attachments.